Modules/SubModule_Import/xASL_imp_DCM2NII_Subject.m

```diff
@@ -26,240 +22,141 @@
 % 2. Iterate over visits
 % 3. Loop through all sessions
 % 4. Iterate over scans
-% -  1. Initialize variables (scanID, summary_line, first_match)
-% -  2. Convert scan ID to a suitable name and set scan-specific parameters
-% -  3. Minimalistic feedback of where we are
-% -  4. Now pick the matching one from the folder list
-% -  5. Determine input and output paths
-% -  6. Start the conversion if this scan should not be skipped
-% -  7. Store JSON files
-% -  8. In case of a single NII ASL file loaded from PAR/REC, we need to shuffle the dynamics from CCCC...LLLL order to CLCLCLCL... order
-% -  9  Copy single dicom as QC placeholder
-% - 10. Store the summary info so it can be sorted and printed below
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
-% EXAMPLE:     [imPar, summary_lines, PrintDICOMFields, globalCounts, scanNames, dcm2niiCatchedErrors, pathDcmDict] = xASL_imp_DCM2NII_Subject(x, imPar, iSubject, matches, dcm2niiCatchedErrors);
+% EXAMPLE:     [x, imPar, PrintDICOMFields, dcm2niiCatchedErrors] = xASL_imp_DCM2NII_Subject(x, imPar, matches, dcm2niiCatchedErrors);
 % __________________________________
 % Copyright 2015-2021 ExploreASL
 
 
     %% 1. Run DCM2NII for one individual subject
     
-    separatorline = '==============================================================================================';
-    summary_lines = x.modules.import.summary_lines;
-    pathDcmDict = x.modules.import.pathDcmDict;
-    scanNames = x.modules.import.scanNames;
-    globalCounts = x.modules.import.globalCounts;
-    numOf = x.modules.import.numOf;
-    settings = x.modules.import.settings;
-    listsIDs = x.modules.import.listsIDs;
-    subjectID = listsIDs.subjectIDs{iSubject};
+    % We do not iterate over subjects anymore, since this is done in xASL_Iteration now
+    iSubject = find(strcmp(x.SUBJECT,x.SUBJECTS));
+    
+    % Overview subjects
```

Comment:
@MichaelStritt

Reply:
So before there were a lot of existing bugs if the number of scans/sessions and so on differed for subjects/sessions etc.

The idea of the overview sub-structure is to help with that. If you have a nicely working sourceStructure.json the basic structure can be read and you get a nice subject/session/run overview. It's a more interactive or OOP based approach to solve this problem. I agree that this could need a bit more documentation though. Please keep in mind that it's hard to write novels about new code while trying to keep everything together and improve existing stuff.

Reply:
2478212

Reply:
Graphic example

![image](https://user-images.githubusercontent.com/46593074/141331504-5d63b71d-3154-44fa-9f36-e391e9e919cb.png)

Reply:
Exemplary usage

![image](https://user-images.githubusercontent.com/46593074/141331652-6474e28c-4e98-48f0-b019-aba20a8d5d36.png)

Reply:
Very nice, though this starts getting confusing. Firstly, I would move this to `x.modules.import.overview`, as this should not be mixed up with the processing modules. Secondly, this seems to have the same role as `x.SUBJECTS` `x.SESSIONS` etc? So while I may like this overview more, I wonder if it doesn't do something we already have (but in a mixed up way). So then it seems more helpful to create a new issue in which we somehow move all `x.SUBJECTS` `x.SESSIONS` etc into a single field, like this?

Probably this becomes simpler if we move fully to BIDS internally as well, as we won't have double names for the same thing anymore ;) And we would always have all three levels subjects sessions runs for all datasets, always iterating over these three — even if there is only one subject session run; this would make the code easier.

Reply:
We talked about this before. x.SUBJECTS, x.SESSIONS etc. are these symbolic fields which are used all over xASL_Iteration and a million times in the code. We already planned to move them, but this is extremely time consuming and will therefore be done in a future release. The dependencies there are quite complex...